### ISSUE TITLE
make 3rd party filters over http opt-in/disabled by default

### ISSUE THREAD

**Issue by sslizer** [gorhill/uBlock#887]
We are living in a time where downloads over HTTP is quite inexcusable and can potentially put users at risk of having their downloads manipulated for malicious gain.

The 3rd party filters tab has a mixture of sites some over HTTP but thankfully the majority over HTTPS now.  I have been contacting the maintainers and asking if they would consider moving to HTTPS, most are interested, especially now that letsencrypt.org will be going public soon but there have been a worrying few who have no intention of moving to HTTPS and in not understanding HTTPS claiming as their site has never been hacked there is no need.

I would like to propose, in order to increase the security of uBlock Origin users than all 3rd party filters that use HTTP as their delivery mechanism become opt-in/default disabled  instead of the case of a few which are opt-out/enabled by default.

**Comment by gorhill** [comment]
Duplicate of #776. See [my answer](https://github.com/gorhill/uBlock/issues/776#issuecomment-145340096).

**Comment by publicarray** [comment]
I agree https(TLS) is the future (especially with letsencrypt.org). But considering that filter lists are available to the public and that there is no personal data transferred I am perfectly comfortable with unencrypted connections.

>  moving to HTTPS and in not understanding HTTPS claiming as their site has never been hacked there is no need.

You should note that https will not protect you from a hack on a server. TLS only encrypts traffic from your client to the server. This only prevents anyone in-between from snooping on your network traffic.

**Comment by sslizer** [comment]
@publicarray 

> But considering that filter lists are available to the public and that there is no personal data transferred I am perfectly comfortable with unencrypted connections.

It does not have to be personal data, your download of the filter list can be manipulated for malicious gain.

> You should note that https will not protect you from a hack on a server. TLS only encrypts traffic from your client to the server. This only prevents anyone in-between (e.g. ISP's and your network that you are connected to) from snooping on your network traffic.

Yes I am fully aware of that HTTPS can do, my point was the person offering the list did not.

The problem with HTTP is that it allows an attacker to inject or modify your traffic.  So you could end up with a with modified filters that now send you to malicious sites or have you take part in attacks as was seen with the [the great cannon](https://threatpost.com/github-attack-perpetrated-by-chinas-great-cannon-traffic-injection-tool/112109/)  where China manipulated people browsing HTTP sites so that they became part of an attack on github

**Comment by gorhill** [comment]
> you could end up with a with modified filters that now send you to malicious sites

The filters are not used to redirect network requests, they are used to block (or unblock blocked) network requests.
